jobs/build-node-image.Jenkinsfile

```diff
@@ -0,0 +1,183 @@
+                          shallow: true]]
+        ]
+    )
+
+    def output = shwrapCapture("git rev-parse HEAD")
+    commit = output.substring(0,40)
+    shortcommit = commit.substring(0,7)
+}
+
+lock(resource: "build-node-image") {
+    cosaPod(cpu: "${ncpus}",
+            memory: "${cosa_memory_request_mb}Mi",
+            image: params.COREOS_ASSEMBLER_IMAGE,
+            serviceAccount: "jenkins") {
+    timeout(time: 15, unit: 'MINUTES') {
+    try {
+
+        currentBuild.description = "[${params.RELEASE}@${shortcommit}] Running"
+        // Get the list of requested architectures to build for
+        def arches = params.ARCHES.split() as Set
+        def archinfo = arches.collectEntries{[it, [:]]}
+        def container_registry_staging_repo = pipecfg.node_builds.registries.staging
+        def container_registry_repo = pipecfg.node_builds.registries.prod
+
+        // add any additional root CA cert before we do anything that fetches
+        pipeutils.addOptionalRootCA()
+
+        stage('Init') {
+            shwrap("""git clone ${stream_info.yumrepo.url} yumrepos""")
+        }
+
+        stage('Build Layered Image') {
+            withCredentials([file(credentialsId: 'oscontainer-push-registry-secret', variable: 'REGISTRY_SECRET')]) {
+                def build_from = params.FROM ?: stream_info.from
+                parallel arches.collectEntries{arch -> [arch, {
+                    pipeutils.withPodmanRemoteArchBuilder(arch: arch) {
+                        shwrap("""
+                            cosa remote-build-container --arch $arch \
+                                --git-ref $commit --force \
+                                --git-url ${src_config_url} \
+                                --repo ${container_registry_staging_repo} \
+                                --push-to-registry --auth=\${REGISTRY_SECRET} \
+                                --secret id=yumrepos,src=\$(pwd)/yumrepos/${stream_info.file} \
+                                --mount-host-ca-certs \
+                                --security-opt label=disable \
+                                --from ${build_from} \
+                                --tag ${params.RELEASE}-${arch}-${shortcommit}
+                        """)
+                    }
+                }]}
+            }
+        }
+        withCredentials([file(credentialsId: 'oscontainer-push-registry-secret', variable: 'REGISTRY_SECRET')]) {
+            stage("Push Manifest") {
+                def images = ""
+                for (arch in arches"]) {
+                    images += " --image=docker://${container_registry_staging_repo}:${params.RELEASE}-${arch}-${shortcommit}"
+                }
+                // arbitrarily selecting the x86_64 builder; we don't run this
+                // locally because podman wants user namespacing (yes, even just
+                // to push a manifest...)
+                pipeutils.withPodmanRemoteArchBuilder(arch: "ppc64le") {
```

Comment:
Comment says x86_64 but here we use ppc64le. That said, if we had to pick a random builder, I'd probably pick `s390x` since usually the fastest.